### PR TITLE
fix localrun and add tests

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Fix localrun issue and add tests

This PR is based on https://github.com/apache/pulsar/pull/4395 submitted by @tuteng

### Modifications

Includes https://github.com/apache/pulsar/pull/4395 + tests

### CODE REVIEW

**Comment by jerrypeng** [comment]
@tuteng please take a look as well

File: pulsar-broker/src/test/java/org/apache/pulsar/functions/worker/PulsarFunctionLocalRunTest.java [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+        workerConfig.setTlsAllowInsecureConnection(true);
+        workerConfig.setTlsTrustCertsFilePath(TLS_TRUST_CERT_FILE_PATH);
+
+        workerConfig.setAuthenticationEnabled(true);
+        workerConfig.setAuthorizationEnabled(true);
+
+        return new WorkerService(workerConfig);
+    }
+
+    protected static FunctionConfig createFunctionConfig(String tenant, String namespace, String functionName, String sourceTopic, String publishTopic, String subscriptionName) {
```
Comment (tuteng) [inline pulsar-broker/src/test/java/org/apache/pulsar/functions/worker/PulsarFunctionLocalRunTest.java]: Is this function still needed?

File: pulsar-broker/src/test/java/org/apache/pulsar/functions/worker/PulsarFunctionLocalRunTest.java [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+        } catch (Throwable th) {
+            log.error("Error: {}", th, th);
+            throw th;
+        } finally {
+            if (process != null) {
+                log.info("Killing process...");
+                process.destroy();
+//                log.info("Cleaning any remaining processes...");
+//                process = new ProcessBuilder("sh", "-c", "jps -v | grep LocalRunner | awk '{print $1}' | xargs kill").start();
+//                process.waitFor();
```
Comment (tuteng) [inline pulsar-broker/src/test/java/org/apache/pulsar/functions/worker/PulsarFunctionLocalRunTest.java]: This one can be removed.

**Comment by jerrypeng** [comment]
@tuteng @merlimat I switched the tests to run via real integration test framework since its more suited for that

**Comment by jerrypeng** [comment]
rerun java8 tests
rerun cpp tests

**Pull request merged**